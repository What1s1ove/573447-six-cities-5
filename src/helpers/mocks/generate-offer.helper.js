import {getRandomItem} from '~/helpers/array/array';
import {getRandomNumber} from '~/helpers/number/number';
import {generateRandomId} from '~/helpers/mocks/generate-random-id.helper';
import {OfferCity, OfferType} from '~/common/enums/enums';

const TITLES = [
  `Beautiful \u0026 luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
];

const OfferConfig = {
  PRICE: {
    MIN: 50,
    MAX: 250,
  },
  RATING: {
    MIN: 0,
    MAX: 5,
    FRACTIONAL_PART_COUNT: 1,
  },
};

const cities = Object.values(OfferCity);
const offerTypes = Object.values(OfferType);

const generateOffer = () => ({
  id: generateRandomId(),
  city: getRandomItem(cities),
  type: getRandomItem(offerTypes),
  title: getRandomItem(TITLES),
  price: getRandomNumber(OfferConfig.PRICE.MIN, OfferConfig.PRICE.MAX),
  rating: getRandomNumber(
      OfferConfig.RATING.MIN,
      OfferConfig.RATING.MAX,
      OfferConfig.RATING.FRACTIONAL_PART_COUNT
  ),
});

export {generateOffer};
