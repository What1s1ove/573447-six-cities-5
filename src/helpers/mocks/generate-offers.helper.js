import {generateOffer} from '~/helpers/mocks/generate-offer.helper';

const generateOffers = (count) => {
  const offers = Array.from(new Array(count), generateOffer);

  return offers;
};

export {generateOffers};
