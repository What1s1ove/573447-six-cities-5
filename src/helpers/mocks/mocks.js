export * from './generate-offer.helper';
export * from './generate-offers.helper';
export * from './generate-random-id.helper';
