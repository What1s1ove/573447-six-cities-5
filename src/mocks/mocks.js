export * from './offers/offers';
export * from './user/user';
export * from './reviews/reviews';
export * from './city/city';
export * from './store/store';
