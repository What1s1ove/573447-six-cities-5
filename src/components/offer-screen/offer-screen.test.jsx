import * as React from 'react';
import {render} from 'enzyme';
import {Provider} from 'react-redux';
import {MemoryRouter} from 'react-router-dom';
import {store, mockedOffer} from '~/mocks/mocks';
import OfferScreen from './offer-screen';

describe(`OfferScreen component`, () => {
  it(`should render OfferScreen component`, () => {
    const component = render(
        <Provider store={store}>
          <MemoryRouter>
            <OfferScreen activeItem={mockedOffer} onActiveItemChange={jest.fn()} />
          </MemoryRouter>
        </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
