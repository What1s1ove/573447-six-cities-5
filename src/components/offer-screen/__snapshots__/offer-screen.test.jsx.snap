// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferScreen component should render OfferScreen component 1`] = `
<div
  class="page"
>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="header__wrapper"
      >
        <div
          class="header__left"
        >
          <a
            class="header__logo-link header__logo-link--active"
            href="/"
          >
            <img
              alt="6 cities logo"
              class="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          class="header__nav"
        >
          <ul
            class="header__nav-list"
          >
            <li
              class="header__nav-item user"
            >
              <a
                class="header__nav-link header__nav-link--profile"
                href="/favorites"
              >
                <div
                  class="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  class="header__user-name user__name"
                >
                  oliver@mail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    class="page__main page__main--property"
  >
    <section
      class="property"
    >
      <div
        class="property__gallery-container container"
      >
        <div
          class="property__gallery"
        >
          <div
            class="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              class="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg"
            />
          </div>
          <div
            class="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              class="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg"
            />
          </div>
          <div
            class="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              class="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg"
            />
          </div>
          <div
            class="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              class="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg"
            />
          </div>
          <div
            class="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              class="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg"
            />
          </div>
          <div
            class="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              class="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg"
            />
          </div>
        </div>
      </div>
      <div
        class="property__container container"
      >
        <div
          class="property__wrapper"
        >
          <div
            class="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            class="property__name-wrapper"
          >
            <h1
              class="property__name"
            >
              The Joshua Tree House
            </h1>
            <button
              class="property__bookmark-button button"
              type="button"
            >
              <svg
                class="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  href="#icon-bookmark"
                />
              </svg>
              <span
                class="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            class="property__rating rating"
          >
            <div
              class="property__stars rating__stars"
            >
              <span
                style="width:80%"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              class="property__rating-value rating__value"
            >
              4.4
            </span>
          </div>
          <ul
            class="property__features"
          >
            <li
              class="property__feature property__feature--entire"
            >
              Private Room
            </li>
            <li
              class="property__feature property__feature--bedrooms"
            >
              1 Bedrooms
            </li>
            <li
              class="property__feature property__feature--adults"
            >
              Max 2 adults
            </li>
          </ul>
          <div
            class="property__price"
          >
            <b
              class="property__price-value"
            >
              €241
            </b>
            <span
              class="property__price-text"
            >
               night
            </span>
          </div>
          <div
            class="property__inside"
          >
            <h2
              class="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              class="property__inside-list"
            >
              <li
                class="property__inside-item"
              >
                Dishwasher
              </li>
              <li
                class="property__inside-item"
              >
                Towels
              </li>
              <li
                class="property__inside-item"
              >
                Washer
              </li>
              <li
                class="property__inside-item"
              >
                Fridge
              </li>
              <li
                class="property__inside-item"
              >
                Breakfast
              </li>
              <li
                class="property__inside-item"
              >
                Laptop friendly workspace
              </li>
              <li
                class="property__inside-item"
              >
                Baby seat
              </li>
              <li
                class="property__inside-item"
              >
                Air conditioning
              </li>
            </ul>
          </div>
          <div
            class="property__host"
          >
            <h2
              class="property__host-title"
            >
              Meet the host
            </h2>
            <div
              class="property__host-user user"
            >
              <div
                class="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
              >
                <img
                  alt="Host avatar"
                  class="property__avatar user__avatar"
                  height="74"
                  src="img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                class="property__user-name"
              >
                Angelina
              </span>
            </div>
            <div
              class="property__description"
            >
              <p
                class="property__text"
              >
                This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.
              </p>
            </div>
          </div>
          <section
            class="property__reviews reviews"
          >
            <h2
              class="reviews__title"
            >
              Reviews ·
              <span
                class="reviews__amount"
              >
                3
              </span>
            </h2>
            <ul
              class="reviews__list"
            >
              <li
                class="reviews__item"
              >
                <div
                  class="reviews__user user"
                >
                  <div
                    class="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      class="reviews__avatar user__avatar"
                      height="54"
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    class="reviews__user-name"
                  >
                    Jack
                  </span>
                </div>
                <div
                  class="reviews__info"
                >
                  <div
                    class="reviews__rating rating"
                  >
                    <div
                      class="reviews__stars rating__stars"
                    >
                      <span
                        style="width:80%"
                      />
                      <span
                        class="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    class="reviews__text"
                  >
                    I stayed here for one night and it was an unpleasant experience.
                  </p>
                  <time
                    class="reviews__time"
                    datetime="2020-10-04T09:29:21.139Z"
                  >
                    October 2020
                  </time>
                </div>
              </li>
              <li
                class="reviews__item"
              >
                <div
                  class="reviews__user user"
                >
                  <div
                    class="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      class="reviews__avatar user__avatar"
                      height="54"
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    class="reviews__user-name"
                  >
                    Corey
                  </span>
                </div>
                <div
                  class="reviews__info"
                >
                  <div
                    class="reviews__rating rating"
                  >
                    <div
                      class="reviews__stars rating__stars"
                    >
                      <span
                        style="width:80%"
                      />
                      <span
                        class="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    class="reviews__text"
                  >
                    The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.
                  </p>
                  <time
                    class="reviews__time"
                    datetime="2020-10-04T09:29:21.139Z"
                  >
                    October 2020
                  </time>
                </div>
              </li>
              <li
                class="reviews__item"
              >
                <div
                  class="reviews__user user"
                >
                  <div
                    class="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      class="reviews__avatar user__avatar"
                      height="54"
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    class="reviews__user-name"
                  >
                    Isaac
                  </span>
                </div>
                <div
                  class="reviews__info"
                >
                  <div
                    class="reviews__rating rating"
                  >
                    <div
                      class="reviews__stars rating__stars"
                    >
                      <span
                        style="width:80%"
                      />
                      <span
                        class="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    class="reviews__text"
                  >
                    The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.
                  </p>
                  <time
                    class="reviews__time"
                    datetime="2020-10-04T09:29:21.139Z"
                  >
                    October 2020
                  </time>
                </div>
              </li>
            </ul>
            <form
              action="#"
              class="reviews__form form"
              method="post"
            >
              <label
                class="reviews__label form__label"
                for="review"
              >
                Your review
              </label>
              <div
                class="reviews__rating-form form__rating"
              >
                <input
                  class="form__rating-input visually-hidden"
                  id="5-stars"
                  name="rating"
                  required=""
                  type="radio"
                  value="5"
                />
                <label
                  class="reviews__rating-label form__rating-label"
                  for="5-stars"
                  title="perfect"
                >
                  <svg
                    class="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      href="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  class="form__rating-input visually-hidden"
                  id="4-stars"
                  name="rating"
                  required=""
                  type="radio"
                  value="4"
                />
                <label
                  class="reviews__rating-label form__rating-label"
                  for="4-stars"
                  title="good"
                >
                  <svg
                    class="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      href="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  class="form__rating-input visually-hidden"
                  id="3-stars"
                  name="rating"
                  required=""
                  type="radio"
                  value="3"
                />
                <label
                  class="reviews__rating-label form__rating-label"
                  for="3-stars"
                  title="notBad"
                >
                  <svg
                    class="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      href="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  class="form__rating-input visually-hidden"
                  id="2-stars"
                  name="rating"
                  required=""
                  type="radio"
                  value="2"
                />
                <label
                  class="reviews__rating-label form__rating-label"
                  for="2-stars"
                  title="badly"
                >
                  <svg
                    class="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      href="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  class="form__rating-input visually-hidden"
                  id="1-stars"
                  name="rating"
                  required=""
                  type="radio"
                  value="1"
                />
                <label
                  class="reviews__rating-label form__rating-label"
                  for="1-stars"
                  title="terribly"
                >
                  <svg
                    class="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      href="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                class="reviews__textarea form__textarea"
                id="review"
                maxlength="300"
                name="review"
                placeholder="Tell how was your stay, what you like and what can be improved"
                required=""
              />
              <div
                class="reviews__button-wrapper"
              >
                <p
                  class="reviews__help"
                >
                  To submit review please make sure to set
                  <span
                    class="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least
                  <b
                    class="reviews__text-amount"
                  >
                    50 characters
                  </b>
                  .
                </p>
                <button
                  class="reviews__submit form__submit button"
                  disabled=""
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </section>
        </div>
      </div>
      <section
        class="property__map map"
      >
        <div
          id="map"
        />
      </section>
    </section>
    <div
      class="container"
    >
      <section
        class="near-places places"
      >
        <h2
          class="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          class="places__list near-places__list"
        >
          <article
            class="cities__place-card place-card"
          >
            <div
              class="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              class="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offer/1"
              >
                <img
                  alt="Place image"
                  class="place-card__image"
                  height="200"
                  src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              class="place-card__info"
            >
              <div
                class="place-card__price-wrapper"
              >
                <div
                  class="place-card__price"
                >
                  <b
                    class="place-card__price-value"
                  >
                    €241
                  </b>
                  <span
                    class="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  class="place-card__bookmark-button button"
                  type="button"
                >
                  <svg
                    class="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      href="#icon-bookmark"
                    />
                  </svg>
                  <span
                    class="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                class="place-card__rating rating"
              >
                <div
                  class="place-card__stars rating__stars"
                >
                  <span
                    style="width:80%"
                  />
                  <span
                    class="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                class="place-card__name"
              >
                <a
                  href="/offer/1"
                >
                  The Joshua Tree House
                </a>
              </h2>
              <p
                class="place-card__type"
              >
                room
              </p>
            </div>
          </article>
          <article
            class="cities__place-card place-card"
          >
            <div
              class="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offer/2"
              >
                <img
                  alt="Place image"
                  class="place-card__image"
                  height="200"
                  src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              class="place-card__info"
            >
              <div
                class="place-card__price-wrapper"
              >
                <div
                  class="place-card__price"
                >
                  <b
                    class="place-card__price-value"
                  >
                    €909
                  </b>
                  <span
                    class="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  class="place-card__bookmark-button button"
                  type="button"
                >
                  <svg
                    class="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      href="#icon-bookmark"
                    />
                  </svg>
                  <span
                    class="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                class="place-card__rating rating"
              >
                <div
                  class="place-card__stars rating__stars"
                >
                  <span
                    style="width:100%"
                  />
                  <span
                    class="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                class="place-card__name"
              >
                <a
                  href="/offer/2"
                >
                  Loft Studio in the Central Area
                </a>
              </h2>
              <p
                class="place-card__type"
              >
                house
              </p>
            </div>
          </article>
          <article
            class="cities__place-card place-card"
          >
            <div
              class="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offer/3"
              >
                <img
                  alt="Place image"
                  class="place-card__image"
                  height="200"
                  src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              class="place-card__info"
            >
              <div
                class="place-card__price-wrapper"
              >
                <div
                  class="place-card__price"
                >
                  <b
                    class="place-card__price-value"
                  >
                    €191
                  </b>
                  <span
                    class="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  class="place-card__bookmark-button button place-card__bookmark-button--active"
                  type="button"
                >
                  <svg
                    class="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      href="#icon-bookmark"
                    />
                  </svg>
                  <span
                    class="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                class="place-card__rating rating"
              >
                <div
                  class="place-card__stars rating__stars"
                >
                  <span
                    style="width:80%"
                  />
                  <span
                    class="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                class="place-card__name"
              >
                <a
                  href="/offer/3"
                >
                  The Pondhouse - A Magical Place
                </a>
              </h2>
              <p
                class="place-card__type"
              >
                house
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>
</div>
`;
